packages/broker/src/plugins/operator/InspectRandomNodeHelper.ts

```diff
@@ -0,0 +1,116 @@
+import { Contract } from "@ethersproject/contracts"
+import { operatorABI } from "@streamr/network-contracts"
+import type { Operator } from "@streamr/network-contracts"
+import { EthereumAddress, Logger, TheGraphClient, toEthereumAddress } from "@streamr/utils"
+import { OperatorServiceConfig } from "./OperatorPlugin"
+import fetch from 'node-fetch'
+
+const logger = new Logger(module)
+
+export class InspectRandomNodeHelper {
+
+    private readonly operatorContractAddress: EthereumAddress
+    private readonly operatorContract: Operator
+    private readonly theGraphClient: TheGraphClient
+
+    constructor(config: OperatorServiceConfig) {
+        this.operatorContractAddress = config.operatorContractAddress
+        const signer = config.signer.connect(config.provider)
+        this.operatorContract = new Contract(config.operatorContractAddress, operatorABI, signer) as unknown as Operator
+        this.theGraphClient = new TheGraphClient({
+            serverUrl: config.theGraphUrl,
+            fetch,
+            logger
+        })
+    }
+
+    async getSponsorshipsOfOperator(operatorAddress: EthereumAddress, requiredBlockNumber: number): Promise<EthereumAddress[]> {
+        
+        const createQuery = (lastId: string, pageSize: number) => {
+            return {
+                query: `
+                    {
+                        operator(id: "${operatorAddress}") {
+                            stakes(where: {id_gt: "${lastId}"}, first: ${pageSize}) {
+                                id
+                                sponsorship {
+                                    id
+                                }
+                            }
+                        }
+                        _meta {
+                            block {
+                            number
+                            }
+                        }
+                    }
+                    `
+            }
+        }
+        const parseItems = (response: { operator: { stakes: { id: string, sponsorship: { id: string } }[] } }): 
+        { id: string, sponsorship: { id: string } }[] => {
+            if (!response.operator) {
+                logger.error('Unable to find operator in The Graph', { operatorContractAddress: this.operatorContractAddress })
+                return []
+            }
+            return response.operator.stakes
+        }
+        this.theGraphClient.updateRequiredBlockNumber(requiredBlockNumber)
+        const queryResult = this.theGraphClient.queryEntities<{ id: string, sponsorship: { id: string } }>(createQuery, parseItems)
+        const sponsorshipIds = new Set<EthereumAddress>()
+        for await (const stake of queryResult) {
+            const sponsorshipId = stake.sponsorship?.id
+            if (sponsorshipId) {
```

Comment:
We have a convention of preferring explicit undefined checks (so that the semantics are more clear)
```suggestion
            if (sponsorshipId !== undefined) {
```

Reply:
Is this check needed? If a steak always belongs to some sponsorship?